### PR TITLE
Fix RabbitMQ StackTrace tag

### PR DESCRIPTION
**Pull request by eerhardt**
We are logging the full exception.ToString() in the exception.stacktrace tag.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2163)

### CODE REVIEW

**Comment by radical** [comment]
Where would this normally get consumed?

File: src/Components/Aspire.RabbitMQ.Client/AspireRabbitMQExtensions.cs [hunk 1]
```diff
@@ -172,7 +172,7 @@ private static IConnection CreateConnection(IConnectionFactory factory, int retr
                 if (connectAttemptActivity is not null)
                 {
                     connectAttemptActivity.AddTag("exception.message", ex.Message);
-                    connectAttemptActivity.AddTag("exception.stacktrace", ex.ToString());
+                    connectAttemptActivity.AddTag("exception.stacktrace", ex.StackTrace);
```
Comment (JamesNK) [inline src/Components/Aspire.RabbitMQ.Client/AspireRabbitMQExtensions.cs:175]:
```suggestion
                    // OTEL's exception.stacktrace attribute includes the type and message.
                    connectAttemptActivity.AddTag("exception.stacktrace", ex.ToString());
```

**Review by JamesNK — DISMISSED** [review]
It's correct now. exception.stacktrace includes the type and message. See https://opentelemetry.io/docs/specs/semconv/attributes-registry/exception/

The attribute is badly named but that's the standard 🤷

File: src/Components/Aspire.RabbitMQ.Client/AspireRabbitMQExtensions.cs (hunk 1 above)
Comment (eerhardt) [reply src/Components/Aspire.RabbitMQ.Client/AspireRabbitMQExtensions.cs]: Thanks for catching this. This is a bad name, so adding a comment makes a lot of sense.

**Review by JamesNK — APPROVED** [review]

**Pull request merged**